src/history/hs.c

```diff
@@ -826,6 +873,26 @@ __wt_hs_cursor_position(WT_SESSION_IMPL *session, WT_CURSOR *cursor, uint32_t bt
     /* NOTREACHED */
 }
 
+/*
+ * __hs_save_read_timestamp --
+ *     Save the currently running transaction's read timestamp into a variable.
+ */
+static void
+__hs_save_read_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *saved_timestamp)
```

Comment:
Maybe overkill but I'm using the same pattern that we use for `__hs_set_isolation` and `__hs_restore_isolation`.